packages/stress/src/mode/chat/schema/executeCommand.ts

```diff
@@ -0,0 +1,80 @@
+import { StressClient } from '../../../utils/stressClient'
+import { ChatConfig } from '../../common/types'
+import { Command } from './testPlanSchema'
+import { getLogger } from '../../../utils/logger'
+import { joinSpace } from './commands/joinSpaceCommand'
+import { check } from '@river-build/dlog'
+import { mintMemberships } from './commands/mintMembershipCommand'
+import { sendChannelMessage } from './commands/sendChannelMessageCommand'
+import { expectChannelMessages } from './commands/expectChannelMessageCommand'
+
+export async function executeCommand(
+    command: Command,
+    commandId: string, // For logging / error reporting
+    chatConfig: ChatConfig,
+    clients: StressClient[],
+) {
+    const logger = getLogger('stress:run', { function: 'executeCommand' })
+    logger.info({ command }, '--------------- executing command ---------------')
+
+    const range = (start: number, end: number) =>
+        Array.from({ length: end - start }, (v, k) => k + start)
+    // run on specified clients, or if unspecified, run on all clients assigned to this process
+    const targetClientIndices =
+        command.targetClients ??
+        range(chatConfig.localClients.startIndex, chatConfig.localClients.endIndex)
+    const targetClients = clients.filter((client) =>
+        targetClientIndices.includes(client.clientIndex),
+    )
+
+    if (targetClients.length === 0) {
+        logger.debug({ commandId }, 'No local clients to execute command')
+        return
+    }
+
+    logger.debug({ commandId, targetClientIndices }, 'Executing command for clients', 'command')
+    type execFn = (client: StressClient, cfg: ChatConfig) => Promise<void>
+    let execute: execFn | undefined = undefined
+
+    switch (command.name) {
+        case 'mintMemberships':
+            {
+                execute = async (client: StressClient, cfg: ChatConfig) => {
```

Comment:
if execFn is updated to have params as 3rd argument, then it's possible to assign it directly. And then it's a bit easier to map even without switch like this:

```typescript
const commandFns : { [key: string]: execFn } = {
  mintMemberships: mintMemberships,
  joinSpace: joinSpace,
  fooBar: fooBar,
};
const execute = commandFns[command.name]
```

Reply:
The param shape actually varies by command type - does that work for your suggestion?

Reply:
@texuf also ^ - we're running up against the limitations of my typescript type foo